pants-plugins/internal_plugins/test_lockfile_fixtures/rules.py

```diff
@@ -0,0 +1,270 @@
+        pytest.config_request(config_file_dirs),
+    )
+    pytest_runner_pex, config_files = await MultiGet(pytest_runner_pex_get, config_files_get)
+
+    pytest_config_digest = config_files.snapshot.digest
+
+    input_digest = await Get(
+        Digest,
+        MergeDigests(
+            (
+                prepared_sources.source_files.snapshot.digest,
+                pytest_config_digest,
+            )
+        ),
+    )
+
+    extra_env = {
+        "PEX_EXTRA_SYS_PATH": ":".join(prepared_sources.source_roots),
+        **test_extra_env.env,
+    }
+
+    process = await Get(
+        Process,
+        VenvPexProcess(
+            pytest_runner_pex,
+            argv=[*prepared_sources.source_files.files],
+            extra_env=extra_env,
+            input_digest=input_digest,
+            output_files=("tests.json",),
+            description="Collect test lockfile requirements from all tests.",
+            level=LogLevel.DEBUG,
+        ),
+    )
+
+    result = await Get(ProcessResult, Process, process)
+    digest_contents = await Get(DigestContents, Digest, result.output_digest)
+    raw_config_data = json.loads(digest_contents[0].content)
+
+    configs = []
+    for item in raw_config_data:
+        config = JVMLockfileFixtureConfig(
+            definition=JVMLockfileFixtureDefinition.from_kwargs(item["kwargs"]),
+            test_file_path=item["test_file_path"],
+        )
+        configs.append(config)
+
+    return CollectedJVMLockfileFixtureConfigs(configs)
+
+
+@rule
+async def gather_lockfile_fixtures() -> RenderedJVMLockfileFixtures:
+    configs = await Get(CollectedJVMLockfileFixtureConfigs, CollectFixtureConfigsRequest())
+    rendered_fixtures = []
+    for config in configs:
+        artifact_reqs = ArtifactRequirements(
+            [ArtifactRequirement(coordinate) for coordinate in config.definition.coordinates]
+        )
+        lockfile = await Get(CoursierResolvedLockfile, ArtifactRequirements, artifact_reqs)
+        serialized_lockfile = JVMLockfileMetadata.new(artifact_reqs).add_header_to_lockfile(
+            lockfile.to_serialized(),
+            regenerate_command=f"{bin_name()} {InternalGenerateTestLockfileFixturesSubsystem.name}",
+            delimeter="#",
+        )
+
+        lockfile_path = os.path.join(
+            os.path.dirname(config.test_file_path), config.definition.lockfile_rel_path
+        )
+        rendered_fixtures.append(
+            RenderedJVMLockfileFixture(
+                content=serialized_lockfile,
+                path=lockfile_path,
+            )
+        )
+
+    return RenderedJVMLockfileFixtures(rendered_fixtures)
+
+
+class InternalGenerateTestLockfileFixturesSubsystem(GoalSubsystem):
+    name = "internal-generate-test-lockfile-fixtures"
+    help = "[Internal] Generate test lockfile fixtures for Pants tests."
+
+
+class InternalGenerateTestLockfileFixturesGoal(Goal):
+    subsystem_cls = InternalGenerateTestLockfileFixturesSubsystem
```

Comment:
It looks like this needs to be its own Goal because it needs to produce multiple lockfiles, and the `generate-lockfiles` infrastructure only allows each plugin to generate one. I'm not sure that it makes sense to change that, but maybe @Eric-Arellano knows better?